vyper/venom/passes/machinery/inst_updater.py

```diff
@@ -55,19 +54,23 @@ def update(
             if isinstance(op, IRVariable):
                 self.dfg.add_use(op, inst)
 
+        old_outputs = inst.get_outputs()
+
         if opcode in NO_OUTPUT_INSTRUCTIONS:
-            if inst.output is not None:
+            for output in old_outputs:
                 assert new_output is None
-                assert len(uses := self.dfg.get_uses(inst.output)) == 0, (inst, uses)
-                self.dfg.remove_producing_instruction(inst.output)
-                inst.output = None
+                assert len(uses := self.dfg.get_uses(output)) == 0, (inst, uses)
+                self.dfg.remove_producing_instruction(output)
+            inst.set_outputs([])
         else:
             # new_output is None is sentinel meaning "no change"
-            if new_output is not None and new_output != inst.output:
-                if inst.output is not None:
-                    self.dfg.remove_producing_instruction(inst.output)
+            if new_output is not None:
+                old_primary = old_outputs[0] if len(old_outputs) > 0 else None
+                if old_primary is not None and old_primary != new_output:
+                    self.dfg.remove_producing_instruction(old_primary)
                 self.dfg.set_producing_instruction(new_output, inst)
-                inst.output = new_output
+                remaining_outputs = old_outputs[1:] if len(old_outputs) > 1 else []
+                inst.set_outputs([new_output, *remaining_outputs])
```

Comment:
this change is a bit smelly... do we ever do this update when old_outputs has length larger than 1? i would rather assert `len(old_outputs) <= 1`